test/test_instruction.py

```diff
@@ -187,3 +188,18 @@ def test_set_name_ideal():
     assert (instr.ideal == GATES["ry90p"]).all()
     instr.set_name("crzp")
     assert (instr.ideal == GATES["crzp"]).all()
+
+
+@pytest.mark.unit
+def test_correct_bloch_rotation_direction():
+    # makes sure that the rotations on the bloch sphere are in the right direction
+    GATE_NAME = 'ry90p[0]'
+    exp = create_experiment()
+    exp.compute_propagators()
+    # TODO - remove this line after the ideal updating bug gets fixed...
+    exp.pmap.instructions[GATE_NAME].set_ideal(None)
+
+    ideal_gate = exp.pmap.instructions[GATE_NAME].get_ideal_gate(dims=[3])
+    propagator = exp.propagators[GATE_NAME].numpy()
+    # not equal to one because of imperfections in the propagation
+    assert unitary_infid(ideal_gate, propagator, dims=[3]).numpy()[0] < 0.05
```

Comment:
We try to use the numpy `assert_almost_equal` with an explicit threshold for numerical comparisons from now on.

Reply:
👍 done